pkg/reconciler/pipeline/dag/dag_test.go

```diff
@@ -157,23 +157,16 @@ func TestBuild_JoinMultipleRoots(t *testing.T) {
 	b := v1beta1.PipelineTask{Name: "b"}
 	c := v1beta1.PipelineTask{Name: "c"}
 	xDependsOnA := v1beta1.PipelineTask{
```

Comment:
nice to have: There are two kinds of dependencies before - using From and using RunAfter. In this PR, we are getting rid of From. So, RunAfter is the only option. Can we simplify these tests instead of just replacing each instance of From with RunAfter?

Reply:
Indeed - another option could be to use a different kind of dependency - replace the `From` with a result -> param type of dependency.

Reply:
Thanks Dibyo and Andrea, I modified to only have 2 parallel tasks using `runAfter` with another one using result of both.

Reply:
yeah that's a good point - do we have result based dependencies captured in dag?

Reply:
Thanks both, I replaced all the dependencies that previously based on resources to be using result-params.